pynest/examples/brette_et_al_2007/cuba_stdp.py

```diff
@@ -0,0 +1,262 @@
+functions). Excitatory-excitatory synapses implement multiplicative STDP.
+
+This is Benchmark 5 of the FACETS simulator review (Brette et al. 2007):
+- Neuron model: integrate-and-fire (i&f)
+- Synapse model: STDP-current (excitatory-excitatory), static-current (others)
+- Synapse time course: exponential
+- Spike times: grid-constrained
+
+This benchmark demonstrates the capabilities of NEST in simulating large
+networks with heterogeneous synaptic dynamics.
+
+References
+~~~~~~~~~~
+
+.. [1] Brunel N. 2000. Dynamics of sparsely connected networks of excitatory
+       and inhibitory spiking neurons. Journal of Computational Neuroscience.
+       8:183-208.
+       https://doi.org/10.1023/A:1008925309027
+
+.. [2] Brette R, Rudolph M, Carnevale T, Hines M, Beeman D, Bower JM, et al.
+       2007. Simulation of networks of spiking neurons: a review of tools and
+       strategies. Journal of Computational Neuroscience. 23(3):349-398.
+       https://doi.org/10.1007/s10827-007-0038-6
+
+"""
+
+import time
+
+import nest
+import numpy as np
+
+###############################################################################
+# Set benchmark parameters
+
+virtual_processes = 1  # number of virtual processes to use
+simtime = 2000.0  # simulated time [ms]
+dt = 0.1  # simulation step [ms]
+
+NE = 9000  # number of excitatory neurons
+NI = 2250  # number of inhibitory neurons
+
+model = "iaf_psc_exp"  # neuron model
+model_params = {
+    "E_L": 0.0,  # resting membrane potential [mV]
+    "V_m": 0.0,  # initial membrane potential [mV] (randomly initialized below)
+    "V_th": 20.0,  # threshold [mV]
+    "V_reset": 0.0,  # reset potential [mV]
+    "C_m": 250.0,  # capacity of the membrane [pF]
+    "tau_m": 20.0,  # membrane time constant [ms]
+}
+
+mean_potential = 7.0  # mean initial membrane potential [mV]
+sigma_potential = 5.0  # standard deviation of initial membrane potential [mV]
+
+epsilon = 0.1  # connection probability
+delay = 1.5  # synaptic delay [ms]
+JE = 35.0  # peak amplitude of PSC [pA]
+g = 5.0  # relative strength of inhibitory connections
+
+plastic_synapses = True
+stdp_params = {
+    "delay": 1.5,
+    "alpha": 1.05,  # STDP asymmetry parameter
+    "weight": JE,
+    "Wmax": 2.0 * JE,  # max strength of E->E synapses [pA]
+}
+
+sigma_w = 3.0  # initial standard deviation of E->E synapses [pA]
+
+stimulus_params = {
+    "rate": 900.0 * 4.5,  # rate of initial poisson stimulus [Hz]
+}
+
+recorder_params = {
+    "record_to": "ascii",
+    "label": "cuba_stdp",
+}
+
+Nrec = 500  # number of neurons per population to record from
+
+###############################################################################
+# Compute rate function
+
+
+def compute_rate(spike_recorder, n_rec, simtime, num_processes=1):
+    """Compute average firing rate per neuron from spike recorder."""
+    n_spikes = spike_recorder.n_events
+    n_neurons = n_rec / num_processes
+    rate = n_spikes / (n_neurons * simtime) * 1000.0  # convert to Hz
```

Comment:
Maybe this function can be imported from the`brette_et_al_2007_benchmark.py` file, where it is better documented to avoid duplication and thus sources of errors.